### PR TITLE
Reset Operator timers and clean up

### LINKED ISSUE #432: Performance drop at operator recall
It looks like calling back a predefined operator comes with big perfomance loss. This is the log from `gradient_test` that calls `op_fwd()` multiple times:
```
=========================================================================================
Allocating memory for rec_coords ((60, 2))
Allocating memory for rec ((397, 60))
Allocating memory for u ((3, 84, 94))
=========================================================================================
Section section_1<396,1> with OI=0.80 computed in 0.000 s [0.34 GFlops/s]
Section section_2<396,60> with OI=1.54 computed in 0.001 s [1.27 GFlops/s]
Section main<396,82,92> with OI=1.91 computed in 0.026 s [3.66 GFlops/s, 0.00 GPts/s]
=========================================================================================
Allocating memory for rec_coords ((60, 2))
Allocating memory for rec ((397, 60))
Allocating memory for u ((3, 84, 94))
=========================================================================================
Section section_1<396,1> with OI=0.80 computed in 0.000 s [0.22 GFlops/s]
Section section_2<396,60> with OI=1.54 computed in 0.001 s [0.85 GFlops/s]
Section main<396,82,92> with OI=1.91 computed in 0.040 s [2.38 GFlops/s, 0.00 GPts/s]
=========================================================================================
Allocating memory for rec_coords ((60, 2))
Allocating memory for rec ((397, 60))
Allocating memory for u ((3, 84, 94))
=========================================================================================
Section section_1<396,1> with OI=0.80 computed in 0.000 s [0.17 GFlops/s]
Section section_2<396,60> with OI=1.54 computed in 0.001 s [0.64 GFlops/s]
Section main<396,82,92> with OI=1.91 computed in 0.053 s [1.79 GFlops/s, 0.00 GPts/s]
=========================================================================================
Allocating memory for rec_coords ((60, 2))
Allocating memory for rec ((397, 60))
Allocating memory for u ((3, 84, 94))
=========================================================================================
Section section_1<396,1> with OI=0.80 computed in 0.000 s [0.13 GFlops/s]
Section section_2<396,60> with OI=1.54 computed in 0.002 s [0.51 GFlops/s]
Section main<396,82,92> with OI=1.91 computed in 0.068 s [1.42 GFlops/s, 0.00 GPts/s]
=========================================================================================
Allocating memory for rec_coords ((60, 2))
Allocating memory for rec ((397, 60))
Allocating memory for u ((3, 84, 94))
=========================================================================================
Section section_1<396,1> with OI=0.80 computed in 0.000 s [0.11 GFlops/s]
Section section_2<396,60> with OI=1.54 computed in 0.002 s [0.43 GFlops/s]
Section main<396,82,92> with OI=1.91 computed in 0.080 s [1.19 GFlops/s, 0.00 GPts/s]
=========================================================================================
Allocating memory for rec_coords ((60, 2))
Allocating memory for rec ((397, 60))
Allocating memory for u ((3, 84, 94))
=========================================================================================
Section section_1<396,1> with OI=0.80 computed in 0.000 s [0.10 GFlops/s]
Section section_2<396,60> with OI=1.54 computed in 0.002 s [0.37 GFlops/s]
Section main<396,82,92> with OI=1.91 computed in 0.093 s [1.03 GFlops/s, 0.00 GPts/s]
=========================================================================================
Allocating memory for rec_coords ((60, 2))
Allocating memory for rec ((397, 60))
Allocating memory for u ((3, 84, 94))
=========================================================================================
Section section_1<396,1> with OI=0.80 computed in 0.000 s [0.09 GFlops/s]
Section section_2<396,60> with OI=1.54 computed in 0.003 s [0.33 GFlops/s]
Section main<396,82,92> with OI=1.91 computed in 0.104 s [0.92 GFlops/s, 0.00 GPts/s]
=========================================================================================
```

This is a tiny problem and already shows quite a lot. For larger problem:
```
Applying Forward
IntelCompiler: compiled /tmp/52643.admin.default.domain/devito-ENHxZJ/1dc3228d78d0d2f20afda55cbfcc654d144a5c91.c [1.52 s]
=========================================================================================
Section section_1<4210,1> with OI=0.96 computed in 0.002 s [0.17 GFlops/s]
Section section_2<4210,641601> with OI=2.10 computed in 12.880 s [17.62 GFlops/s]
Section main<4210,838,838,267> with OI=2.64 computed in 379.144 s [89.53 GFlops/s, 2.08 GPts/s]
=========================================================================================
=========================================================================================
Section section_1<527,1> with OI=0.96 computed in 0.005 s [0.01 GFlops/s]
Section section_2<527,641601> with OI=2.10 computed in 26.135 s [1.09 GFlops/s]
Section main<527,838,838,267> with OI=2.63 computed in 896.420 s [4.74 GFlops/s, 0.11 GPts/s]
=========================================================================================
```

### PR DESCRIPTION
**Pull request by FabioLuporini**
fixes #432 

@mloubout before landing this, I'd like you to do some advanced beta testing to double check that the reported execution times still make sense. They do to me, but it's so easy to get confused...

### CODE REVIEW

File: devito/operator.py [hunk 1]
```diff
@@ -485,12 +428,82 @@ def _profile_output(self, arguments):
...
         """Introduce C-level profiling nodes within the Iteration/Expression tree."""
-        nodes, profiler = create_profile(nodes)
+        nodes, profiler = create_profile('timers', nodes)
         self._globals.append(profiler.cdef)
-        parameters.append(Object(profiler.varname, profiler.dtype, profiler.setup()))
+        parameters.append(Object(profiler.name, profiler.dtype, profiler.new))
         return nodes, profiler
 
 
+# Functions collecting information from a bag of expressions
```
Comment (FabioLuporini) [inline devito/operator.py]: Note for reviewers: this whole block of code is just stuff being moved, nothing new added

**Comment by mloubout** [comment]
Looks like it does the right thing.

**Comment by FabioLuporini** [comment]
OK, merging...

**Comment by FabioLuporini** [comment]
Merged.

**Pull request merged**